waspc/test/AppSpec/ValidTest.hs

```diff
@@ -41,6 +44,50 @@ spec_AppSpecValid = do
                          "You have more than one 'app' declaration in your Wasp app. You have 2."
                      ]
 
+    describe "'waspVersion' validation" $ do
+      describe "should validate 'waspVersion' format" $ do
+        it "returns no error if waspVersion is compatible" $ do
+          ASV.validateAppSpec basicAppSpec `shouldBe` []
+
+        it "returns an error if 'waspVersion' has an incorrect format" $ do
+          let basicAppWithInvalidWaspVersionFormat = basicApp {AS.App.wasp = AS.Wasp.Wasp {AS.Wasp.version = "0.5.2"}}
+          let basicAppDeclWithInvalidWaspVersionFormat =
+                AS.Decl.makeDecl "TestApp" basicAppWithInvalidWaspVersionFormat
```

Comment:
I would just inline this into `AS.decls = [<inline here>]`, I don't think you get much with this extra name.